jxl/src/frame/quant_weights.rs

```diff
@@ -827,18 +865,31 @@ impl DequantMatrices {
         &self.inv_table[self.table_offsets[quant_kind as usize * 3 + c]..]
     }
 
-    pub fn decode(br: &mut BitReader) -> Result<Self> {
+    pub fn decode(
+        header: &FrameHeader,
+        lf_global: &LfGlobalState,
+        br: &mut BitReader,
+    ) -> Result<Self> {
         let all_default = br.read(1)? == 1;
         let mut encodings = Vec::with_capacity(NUM_QUANT_TABLES);
         if all_default {
             for _ in 0..NUM_QUANT_TABLES {
                 encodings.push(QuantEncoding::Library { predefined: 0 })
             }
         } else {
-            for (&required_size_x, required_size_y) in
-                Self::REQUIRED_SIZE_X.iter().zip(Self::REQUIRED_SIZE_Y)
+            for (i, (&required_size_x, required_size_y)) in Self::REQUIRED_SIZE_X
+                .iter()
+                .zip(Self::REQUIRED_SIZE_Y)
+                .enumerate()
```

Comment:
(Or should this perhaps just be `for i in 0..NUM_QUANT_TABLES` at this stage?)

Reply:
(either is fine)